### DISCUSSION TITLE
HomeSpan on Uno R4

### DISCUSSION THREAD

**Discussion by Fazmat7** [HomeSpan/HomeSpan#696] [Q&A] [answered]
Hi, I'm trying to automate my blinds and integrate it into my HomeKit network and have an Arduino uno R4. I understand that HomeSpan is designed to work with ESP boards but the r4 uses an ESP32-S3 microcontroller chip and was wondering if it was possible to code the chip with HomeSpan so that the board could output commands to the stepper motors?

Any help is appreciated.

Thanks

**Comment by HomeSpan** [comment, marked as answer]
I don't think this will work.  According to the Arduino docs on the R4, the ESP32-S3 is a *support processor* used for serial/WiFi/BLE connectivity.  It is flashed with Arduino-specific code that communicates with the main processor on the R4 (the one that is used to run Arduino programs).  The R4 is not designed to provide direct end-user access to the ESP32-S3 to run your own sketches on that processor.

Of course there are always workarounds and Arduino does say you can flash your own code onto the ESP32-S3 if you really want to, but unless that code also does all the things that the Arduino needs it to, such as linking the main processor through the ESP32-S3 to serial, the Arduino R4 will not function.

**Reply by Fazmat7** [reply]
I understand, thank you very much for that information and help. Time to buy an ESP32 I guess!
